taipy/gui/data/array_dict_data_accessor.py

```diff
@@ -33,14 +33,14 @@ def __get_dataframe(self, value: t.Any) -> t.Union[t.List[pd.DataFrame], pd.Data
             types = {type(x) for x in value}
             if len(types) == 1:
                 type_elt = next(iter(types), None)
-                if type_elt == list:
+                if type_elt is list:
                     lengths = {len(x) for x in value}
                     return (
                         pd.DataFrame(value)
                         if len(lengths) == 1
                         else [pd.DataFrame({f"{i}/0": v}) for i, v in enumerate(value)]
                     )
-                elif type_elt == dict:
+                elif type_elt is dict:
                     return [pd.DataFrame(v) for v in value]
                 elif type_elt == _MapDict:
```

Comment:
why not is ?

Reply:
These are Ruff auto changes... I don't know the precise rule.

Reply:
😶‍🌫️🤔😮😱

Reply:
This was the initial error fixed by ruff:
https://docs.astral.sh/ruff/rules/type-comparison/

Reply:
Does not help understand why it does not complain about `type_elt == _MapDict:`

Reply:
`is` is more efficient but `==` is valid too